Mathlib/Algebra/Ring/BooleanRing.lean

```diff
@@ -46,7 +47,7 @@ variable {α β γ : Type*}
 /-- A Boolean ring is a ring where multiplication is idempotent. -/
 class BooleanRing (α) extends Ring α where
   /-- Multiplication in a boolean ring is idempotent. -/
-  mul_self : ∀ a : α, a * a = a
+  mul_self : ∀ a : α, IsIdempotentElem a
```

Comment:
```suggestion
  isIdempotentElem (a : α) : IsIdempotentElem a
```
and it would be good to restore `mul_self` as a lemma with the statement it had before